src/PortingAssistant.Client.Analysis/Utils/CompatibilityCheckerHelper.cs

```diff
@@ -0,0 +1,318 @@
+                            {
+                                CodeEntityDetails = codeEntity,
+                                CompatibilityResults = new Dictionary<string, CompatibilityResult>
+                                {
+                                    {
+                                        targetFramework, new CompatibilityResult() { Compatibility = Model.Compatibility.UNKNOWN, CompatibleVersions = new List<string>() }
+                                    }
+                                },
+                                Recommendations = new Model.Recommendations
+                                {
+                                    RecommendedActions = new List<RecommendedAction>() {
+
+                                        new RecommendedAction() {
+                                            RecommendedActionType = Model.RecommendedActionType.NoRecommendation,
+                                            Description = null,
+                                            TextSpan = null,
+                                            TargetCPU = null,
+                                            TextChanges = null
+                                        }
+                                    },
+                                    RecommendedPackageVersions = new List<string>()
+                                }
+                            };
+                        }
+
+                        ApiAnalysisResult apiAnalysisResult = RetrieveApiAnalysisResultFromCompatibilityCheckResponse(compatibilityCheckerResponse, targetFramework, codeEntity, package);
+                        if (apiAnalysisResult == null )
+                        {
+                            
+                            var sdkPackage = new Compatibility.Common.Model.PackageVersionPair
+                            {
+                                PackageId = package.PackageId, Version = package.Version,
+                                PackageSourceType = Compatibility.Common.Model.PackageSourceType.SDK
+                            };
+
+                            apiAnalysisResult = RetrieveApiAnalysisResultFromCompatibilityCheckResponse(compatibilityCheckerResponse, targetFramework, codeEntity, sdkPackage);
+                        }
+
+                        if (apiAnalysisResult == null )
+                        {
+                            var sdkPackage = new Compatibility.Common.Model.PackageVersionPair
+                            {
+                                PackageId = codeEntity.Namespace,
+                                Version = "0.0.0",
+                                PackageSourceType = Compatibility.Common.Model.PackageSourceType.SDK
+                            };
+
+                            apiAnalysisResult = RetrieveApiAnalysisResultFromCompatibilityCheckResponse(compatibilityCheckerResponse, targetFramework, codeEntity, sdkPackage);
+                            
+                        }
```

Comment:
Will the 2nd `if` statement ever be true? In the first one, we are checking if `apiAnalysisResult == null` and assigning a value, making `apiAnalysisResult` not null. The 2nd `if` checks the same condition and it will always be false.

Is my understanding incorrect?

Reply:
this is a super tricky logic .  one package can either be SDK or package.  somehow we have seem SDK with non 0.0.0 version.  So I just added this logic incase we miss anything.